ledger/metrics/src/test/lib/scala/com/daml/metrics/api/testing/TestingInMemoryMetrics.scala

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics.api.testing
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.{AtomicInteger, AtomicLong, AtomicReference}
+
+import com.daml.metrics.api.MetricHandle.{Counter, Factory, Gauge, Histogram, Meter, Timer}
+import com.daml.metrics.api.testing.TestingMetrics.{
+  TestingInMemoryCounter,
+  TestingInMemoryGauge,
+  TestingInMemoryHistogram,
+  TestingInMemoryMeter,
+  TestingInMemoryTimer,
+}
+import com.daml.metrics.api.{MetricHandle, MetricName, MetricsContext}
+import com.daml.scalautil.Statement.discard
+
+import scala.collection.concurrent.{TrieMap, Map => ConcurrentMap}
+
+trait TestingInMemoryMetricsFactory extends Factory {
+
+  override def timer(name: MetricName)(implicit
+      context: MetricsContext
+  ): MetricHandle.Timer = TestingInMemoryTimer(context)
+
+  override def gauge[T](name: MetricName, initial: T)(implicit
+      context: MetricsContext
+  ): MetricHandle.Gauge[T] = TestingInMemoryGauge(context)
+
+  override def gaugeWithSupplier[T](
+      name: MetricName,
+      gaugeSupplier: () => T,
+  )(implicit context: MetricsContext): Unit = ()
+
+  override def meter(name: MetricName)(implicit
+      context: MetricsContext
+  ): MetricHandle.Meter = TestingInMemoryMeter(context)
+
+  override def counter(name: MetricName)(implicit
+      context: MetricsContext
+  ): MetricHandle.Counter = TestingInMemoryCounter(context)
+
+  override def histogram(name: MetricName)(implicit
+      context: MetricsContext
+  ): MetricHandle.Histogram = TestingInMemoryHistogram(context)
+
+}
+
+object TestingInMemoryMetricsFactory extends TestingInMemoryMetricsFactory
+
+object TestingMetrics {
```

Comment:
```suggestion
private[testing] object TestingMetrics {
```
Or, alternatively, I'd suggest simply moving all of these definitions into the `InMemoryMetricsFactory` object.